src/signals/incident-management/components/FilterForm/index.js

```diff
@@ -123,19 +123,27 @@ const FilterForm = ({ filter, onCancel, onClearFilter, onSaveFilter, onSubmit, o
   const dateFrom = state.options.created_after && new Date(state.options.created_after);
   const dateBefore = state.options.created_before && new Date(state.options.created_before);
 
+  // Store ref to filter object (used in onSubmitForm).
```

Comment:
Not sure this is the best solution. You store here a part of the reducer state in a ref. It doesn't fill right. If this has to be the solution, I would split the state.filter from the state.options in two separate state structures. In that way would be easier to understand what happens here.